### PR TITLE
DAOS-17338 object: optimize sgl handling to reduce iov buffer count

### PR DESCRIPTION
**Pull request by wangshilong**
This update modifies obj_sgls_dup and obj_dup_sgls_free to merge small or
fragmented IOV buffers, reducing the scatter-gather list (SGL) IOV count and mitigating resource exhaustion during network bulk transfers.

1. IOV Buffer Merging Logic: Buffers smaller than 64 bytes are now merged into a larger contiguous buffer. Sequential buffers exceeding 64 entries (each ≤4KB) are consolidated into a single buffer to minimize IOV count. Original data is copied into the merged buffer, preserving content integrity.

2. Memory Optimization: No additional memory allocation occurs if no merging is required. For write operations, merged cases incur one memory allocation and copy. For fetch operations, two copies are performed to ensure unread regions remain unmodified (critical for CI/test validation).

Reduces SGL memory fragmentation and IOV buffer count, improving bulk transfer efficiency. Addresses edge cases where excessive IOV entries could exhaust network-layer resources.

### Steps for the author:

* [ ] Commit message follows the [guidelines](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Appropriate [Features or Test-tag](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags) pragmas were used.
* [ ] Appropriate [Functional Test Stages](https://daosio.atlassian.net/wiki/spaces/DC/pages/12147556353/CI+Functional+Test+Stages) were run.
* [ ] At least two positive code reviews including at least one code owner from each category referenced in the PR.
* [ ] Testing is complete. If necessary, forced-landing label added and a reason added in a comment.

#### After all prior steps are complete:
* [ ] Gatekeeper requested (daos-gatekeeper added as a reviewer).

### CODE REVIEW

**Comment by daosbuild3** [comment]
Test stage Functional on EL 8.8 completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net/job/daos-stack/job/daos/job/PR-16268/2/display/redirect

**Comment by daosbuild3** [comment]
Test stage Functional Hardware Large completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16268/6/execution/node/1335/log

**Comment by daosbuild3** [comment]
Test stage Functional Hardware Large completed with status FAILURE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16268/7/execution/node/1342/log

**Review by NiuYawei — APPROVED** [review]

File: src/object/cli_obj.c [hunk 1]
```diff
@@ -4474,109 +4484,485 @@ obj_size_fetch_cb(const struct dc_object *obj, struct obj_auxi_args *obj_auxi)
...
+
+	if (ctx->merged_bitmaps && ctx->merged_bitmaps[i] &&
+	    isset_range((uint8_t *)ctx->merged_bitmaps[i], j, j))
+		return 1;
+
+	iov = &sg->sg_iovs[j];
+	/* try to merge with left iov */
+	while (merge_idx >= 1) {
+		iov_left = &sg->sg_iovs[merge_idx - 1];
+		if (skip_sgl_iov(update, iov_left)) {
```
Comment (gnailzenh) [inline src/object/cli_obj.c:4534]: This is because users might provider zero-sized buffer?

**Review by gnailzenh — APPROVED** [review]
